### PR TITLE
Add Hommola cospeciation test

### PR DESCRIPTION
**Pull request by tanaes**
Here, we have added a statistical test for host:parasite cospeciation derived from Hommola et al (2009, Molecular Biology and Evolution). This is added as a new module called evolve under math.stats.

We are looking for feedback on the organization and naming of the evolve module. We're not quite sure what to add to the top level description for this module, so any other ideas would be appreciated. 

Note: the test itself is using numpy arrays rather than distance matrix objects because we are using the positional relationships between host, parasite, and an interaction matrix rather than the labels.

### CODE REVIEW

**Comment by ebolyen** [comment]
@tanaes Just checking in, any new information?

**Comment by tanaes** [comment]
Just got back to the US, sorry this fell off my radar! I will have an
update in the next day or two.

In the meantime, I haven't heard anything back from the original author of
the code... Have now tried getting in touch via Facebook. Will report back.

-j

On Tue, Jan 20, 2015 at 3:18 PM, Evan Bolyen <notifications@github.com>
wrote:

> @tanaes <https://github.com/tanaes> Just checking in, any new information?
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/scikit-bio/pull/557#issuecomment-70725047>.
>



-- 
"If you hold a cat by the tail you learn things you cannot learn any other
way."
                         --Mark Twain

**Comment by ebolyen** [comment]
No worries! This isn't a fast moving target, I just remembered myself and was curious.

File: skbio/stats/evolve/_hommola.py [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+
+    # now do permutatitons. Initialize index lists of the appropriate size.
+    mp = np.arange(num_pars)
+    mh = np.arange(num_hosts)
+
+    # initialize list of shuffled correlation vals
+    perm_stats = np.empty(permutations)
+
+    if permutations == 0 or np.isnan(r):
+        p_val = np.nan
```
Comment (tanaes) [inline skbio/stats/evolve/_hommola.py]: Done!

**Comment by tanaes** [comment]
OK, I addressed the remaining comments above, rebased on the main branch, and it looks like it passes the Travis build. 

Still no reply from the original author of the code, who I also got tried to track down on Facebook. Maybe I will try the senior author of that paper instead? Not sure how that works as far as copyright is concerned, given that the lead author is the only one credited in the code itself.

**Comment by jairideout** [comment]
Thanks @tanaes! I'm tied up for the next few days with the upcoming QIIME 1.9.0 release. After that I should be available to take a final pass through. Thanks for addressing all of these comments!

I'm not sure how best to proceed with getting permission from the original author of the code. @gregcaporaso how do you think we should move forward with this?

**Comment by gregcaporaso** [comment]
I think sending the senior author a message next is the best option.
Thanks!

**Comment by jairideout** [comment]
@tanaes just checking in to see if you heard back from the senior author yet. Once we have a path forward, I'll prioritize getting this merged and will take care of rebasing/merge conflicts/anything else remaining. Thanks for all of your work on this, it'll be exciting to add this functionality to scikit-bio!

**Comment by tanaes** [comment]
Ok, just heard back from the senior author Wally Gilks:

Dear Jon


> Yes, please go ahead!


> Wally
> Wally Gilks
> Department of Statistics
> School of Mathematics
> University of Leeds
> Woodhouse Lane
> Leeds
> LS2 9JT
> UK


I hope that works for us! Let me know if there's anything else I can do!

Cheers,
-j

**Comment by Jorge-C** [comment]
That's awesome! It needs to be rebased on top of master (to solve merge conflicts) for a final review and a merge very soon. Thanks for all your work so far @tanaes :)

**Comment by jairideout** [comment]
Thanks for the update @tanaes! I'll take care of the rebasing and make any final changes. Should have time this week to submit a PR containing your commits.

**Comment by tanaes** [comment]
Awesome -- thank you all! Let me know if there's anything else I can do.

-j

On Mon, Mar 16, 2015 at 1:40 PM, Jai Ram Rideout <notifications@github.com>
wrote:

> Thanks for the update @tanaes
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_tanaes&d=BQMCaQ&c=WO-RGvefibhHBZq3fL85hQ&r=HH7jG2Tt8197HNR9p8tbXSGeEwcBM0prLzmrvLXVX5Q&m=J2tAtZ7fkAa8zhlSXaMaY4qUYDUXDV6ccl7b3IP5u3g&s=c4j7_lrEUCYrA1P_qnEJeJoRDhiAjSB1GkJP3RYZsTo&e=>!
> I'll take care of the rebasing and make any final changes. Should have time
> this week to submit a PR containing your commits.
>
> --
> Reply to this email directly or view it on GitHub
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_biocore_scikit-2Dbio_pull_557-23issuecomment-2D81832453&d=BQMCaQ&c=WO-RGvefibhHBZq3fL85hQ&r=HH7jG2Tt8197HNR9p8tbXSGeEwcBM0prLzmrvLXVX5Q&m=J2tAtZ7fkAa8zhlSXaMaY4qUYDUXDV6ccl7b3IP5u3g&s=e_ZYBtyMkA-8IwcqScsBI1OduzJPRReXCShNmZ03r9U&e=>
> .
>



-- 
"If you hold a cat by the tail you learn things you cannot learn any other
way."
                         --Mark Twain

**Comment by jairideout** [comment]
Closing in favor of #877, which contains all commits in this PR.

**Pull request closed without merging**